src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,273 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function gpkgread(fname; layer=1)
+  db = SQLite.DB(fname)
+  assertgpkg(db)
+  geom = gpkgmesh(db, ; layer)
+  attrs = gpkgmeshattrs(db, ; layer)
+  GeoTables.georef(attrs, geom)
+end
+
+function assertgpkg(db)
+  appid = DBInterface.execute(db, "PRAGMA application_id;") |> first |> only
+  userversion = DBInterface.execute(db, "PRAGMA user_version;") |> first |> only
+
+  if !hasgpkgmetadata(db)
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+
+  # Requirement 6: PRAGMA integrity_check returns a single row with the value 'ok'
+  # Requirement 7: PRAGMA foreign_key_check (w/ no parameter value) returns an empty result set
+  if (DBInterface.execute(db, "PRAGMA integrity_check;") |> first |> only != "ok") ||
```

Comment:
I believe using `first` and `only` in the same statement is not necessary. `first` should be enough?